hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java

```diff
@@ -169,7 +169,9 @@ public boolean isShutdown() {
     public void shutdown() {
         isShutdown = true;
         responseThread.interrupt();
-        callIdMap.clear();
+        CleanResourcesTask cleanResourcesTask = new CleanResourcesTask();
+        cleanResourcesTask.run();
```

Comment:
1) Since all connections are closed already at this point, waitForPacketsProcessed logic guaranteed to not run. 
2) We don't such a method to call directly.
3) Assert check is not for to check something and recover. Assert is just to state that it can not be not empty here. If it is , than it means a bug and we need to fix it.